Branch-SDK/src/main/java/io/branch/referral/TrackingController.java

```diff
@@ -22,19 +23,32 @@ public class TrackingController {
     TrackingController(Context context) {
         updateTrackingState(context);
     }
-    
-    void disableTracking(Context context, boolean disableTracking) {
-        if (trackingDisabled != disableTracking) {
-            trackingDisabled = disableTracking;
-            if (disableTracking) {
-                onTrackingDisabled(context);
-            } else {
-                onTrackingEnabled();
+
+    void disableTracking(Context context, boolean disableTracking, @Nullable Branch.TrackingStateCallback callback) {
+        if (trackingDisabled == disableTracking) {
```

Comment:
Can you add a comment why this check is necessary?

Reply:
Ah yeah that naming is confusing. Just added this comment above to clarify.        
`// If the tracking state is already set to the desired state, then return instantly`